### PR TITLE
sle12 update & add-on fix

### PR DESCRIPTION
**Pull request by dzedro**
ref.
https://progress.opensuse.org/issues/7488
SLE12 update
http://10.100.98.90/tests/650

SLE12 update + SDK
http://10.100.98.90/tests/646

SLE12 update + HA+GEO
http://10.100.98.90/tests/649

https://progress.opensuse.org/issues/7486
SDK
http://10.100.98.90/tests/638

### CODE REVIEW

File: tests/installation/upgrade_select.pm [hunk 1]
```diff
@@ -7,33 +7,9 @@ sub run() {
 
     # hardware detection can take a while
     assert_screen "select-for-update", 100;
-    send_key $cmd{"next"}, 1;
+    send_key $cmd{"next"}, 5;
     assert_screen "remove-repository", 30;
-    send_key $cmd{"next"}, 1;
-    if (check_screen('network-not-configured', 5)) {
```
Comment (sysrich) [inline tests/installation/upgrade_select.pm]: I'm pretty sure all these lines you're removing here are needed by openSUSE

https://openqa.opensuse.org/tests/62377/modules/upgrade_select/steps/3

Comment (sysrich) [inline tests/installation/upgrade_select.pm]: And given they're all 'if check_screen's - why not keep them?

File: tests/installation/addon_products_sle.pm [hunk 2]
```diff
@@ -5,11 +5,40 @@ use testapi;
 
 sub run() {
     my $self = shift;
-    assert_screen 'inst-addon', 3;
-    send_key $cmd{"next"}, 1;    # done
 
-    if (check_screen("local-registration-servers", 10)) {
-        send_key $cmd{ok};
```
Comment (sysrich) [inline tests/installation/addon_products_sle.pm]: As with my other comment in upgrade_select - is there any harm keeping this if check_screen?

**Comment by dzedro** [comment]
sorry, should be better now, I have to develop that three brain you mentioned :)

File: tests/installation/upgrade_select.pm [hunk 3]
```diff
@@ -10,30 +10,32 @@ sub run() {
...
+            }
+        }
+        if (check_screen('list-of-online-repositories', 10)) {
+            send_key 'alt-n';
+            record_soft_failure;
+        }
+        # Bug 881107 - there is 2nd license agreement screen in openSUSE upgrade
+        # http://bugzilla.opensuse.org/show_bug.cgi?id=881107
+        # (remove after the bug is closed)
+        if (check_screen('upgrade-li-cense-agreement', 10)) {
```
Comment (coolo) [inline tests/installation/upgrade_select.pm]: I wonder where that strange tag name comes from :)

**Comment by dzedro** [comment]
opensuse
http://10.100.98.90/tests/720/modules/upgrade_select/steps/1

sle12
http://10.100.98.90/tests/649/modules/upgrade_select/steps/1

File: tests/installation/upgrade_select.pm (hunk 3 above)
Comment (nilxam) [inline tests/installation/upgrade_select.pm]: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/139 you merged it :D

**Pull request merged**